src/lib/mlang.js

```diff
@@ -73,72 +75,182 @@ foam.CLASS({
 });
 
 
+/**
+ * Base class for all mLang queries.
+ *
+ * Contains many default implementations of methods.
+ */
 foam.CLASS({
-  package: 'foam.mlang',
-  name: 'Expr'
+  package: 'foam.mlang.predicate',
+  name: 'Expr',
+  abstract: true,
+  // PORT: There should be several interfaces here eventually, eg.
+  // Functor/Projectable/whatever we call f()'s interface.
+  // implements: ['ExprProtocol'],
+
+  methods: [
+    /**
+     * Abstract projection method. Given an object, extracts the value of a
```

Comment:
This isn't always a projection method, in some cases it's a predicate function, in others it's a projection.